packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart

```diff
@@ -0,0 +1,256 @@
+  /// The currencyCode for the [price], e.g USD for U.S. dollars.
+  final String currencyCode;
+
+  /// The object represent the discount period length.
+  ///
+  /// The value must be >= 0.
+  final int numberOfPeriods;
+
+  /// The object indicates how the discount price is charged.
+  final ProductDiscountPaymentMode paymentMode;
+
+  /// The object represents the duration of single subscription period for the discount.
+  ///
+  /// The [subscriptionPeriod] of the discount is independent of the product's [subscriptionPeriod],
+  /// and their units and duration do not have to be matched.
+  final SKProductSubscriptionPeriodWrapper subscriptionPeriod;
+}
+
+/// Dart wrapper around StoreKit's [SKProduct](https://developer.apple.com/documentation/storekit/skproduct?language=objc).
+///
+/// Most of the fields are identical to OBJC SKProduct.
+/// The only difference is instead of the locale object, we only exposed currencyCode for simplicity.
+/// A list of [SKProductWrapper] is returned in the [SKProductRequestMaker]'s [startProductRequest] method, and
+/// should be stored for use when making a payment.
+class SKProductWrapper {
+  SKProductWrapper({
+    @required this.productIdentifier,
+    @required this.localizedTitle,
+    @required this.localizedDescription,
+    @required this.currencyCode,
+    @required this.downloadContentVersion,
+    @required this.subscriptionGroupIdentifier,
+    @required this.price,
+    @required this.downloadable,
+    @required this.downloadContentLengths,
+    @required this.subscriptionPeriod,
+    @required this.introductoryPrice,
+  });
+
+  /// Used for constructing the class with the map passed from the OBJC layer.
+  ///
+  /// The [map] parameter should not be null.
+  SKProductWrapper.fromMap(Map<dynamic, dynamic> map)
+      : productIdentifier = map['productIdentifier'],
+        localizedTitle = map['localizedTitle'],
+        localizedDescription = map['localizedDescription'],
+        currencyCode = map['currencyCode'],
+        downloadContentVersion = map['downloadContentVersion'],
+        subscriptionGroupIdentifier = map['subscriptionGroupIdentifier'],
+        price = map['price'],
+        downloadable = map['downloadable'],
+        downloadContentLengths =
+            List.castFrom<dynamic, int>(map['downloadContentLengths']),
+        subscriptionPeriod = map['subscriptionPeriod'] != null
+            ? SKProductSubscriptionPeriodWrapper.fromMap(
+                map['subscriptionPeriod'].cast<String, dynamic>())
+            : null,
+        introductoryPrice = (map['introductoryPrice'] != null)
+            ? SKProductDiscountWrapper.fromMap(
+                map['introductoryPrice'].cast<String, dynamic>())
+            : null;
+
+  /// The unique identifier of the product.
+  final String productIdentifier,
+
+      /// The localizedTitle of the product.
+      ///
+      /// It is localized based on the current locale.
+      localizedTitle,
```

Comment:
I can't find a specific section about this in the style guide but this style is not typically used in the Flutter codebase, so I'd stay consistent.

Lets replace each of these with e.g `final String localizedTitle;`.

One reason I could think of why that would be better is that when I stare and one of these deep down in the list (e.g downloadContentVersion) I can't easily tell what the type is but have to scroll up.

@Hixie maybe worth mentioning this in the style guide?